### ISSUE TITLE
Periodical

### ISSUE THREAD

**Issue by rmzelle** [citation-style-language/zotero-bits#23]
There have been requests for a "Periodical" item type, but the idea is still rough.

https://github.com/ajlyon/zotero-bits/wiki/PeriodicalType

**Comment by aurelberra** [comment]
I'm reviving this thread, because it seems the problem hasn't been dealt with in more recent versions of Zotero. I also think that a type is sorely needed for whole (edited) issues of a journal.

**Comment by adam3smith** [comment]
that's why the ticket is open. No new item types and fields have been added to Zotero since this ticket was opened.

**Comment by storytracer** [comment]
+1 As a historian, it is the standard case for me that I have entire volumes of newspapers or periodicals in a single PDF file. The "Periodical" item type is essential for me and a most of my colleagues.

**Comment by denismaier** [comment]
So, we can just add a new type `periodical` right to cite the periodical as a whole. Concerning the fields, I think we will only need to add a `frequency` field.

Ok?

**Comment by brandelune** [comment]
It looks like the checkout code does not yet include this periodical type. Am I wrong ?

**Comment by bwiernik** [comment]
I am not sure what you mean by checkout code. This repository does not contain any code. A `periodical` item type has been added to CSL 1.0.2 and will likely be added to a future Zotero version.

**Comment by brandelune** [comment]
I mean that I checked out the code (cloned it), built the app and thought that I'd see a "periodical" type after reading all the issues related to that discussion and did not find it.

**Comment by dstillman** [comment]
Still not clear what code you're talking about, but this entire repo is just for tracking future schema changes. It doesn't have anything to do with current Zotero code.

**Comment by bwiernik** [comment]
This repository has no code. 

The issue tracker was historically a place where changes to item types in CSL and Zotero were discussed. All of the discussed changes have been implemented in CSL. It now serves as a bit of a checklist for the Zotero devs as they update item types and fields in a future Zotero version.

@dstillman It might be a good idea to archive this repository or disable new issues or comments?

**Comment by brandelune** [comment]
I know that this repository has no code, but tracking this "periodical type" issue led me here and since the issue is not closed I allowed myself a comment regarding the fact that that new type is supposed to be implemented somewhere and I just cloned Zotero's code, built it and it is not there. Hence my comment here regarding the fact that the type did not seem to be implemented yet.

**Comment by denismaier** [comment]
> I allowed myself a comment regarding the fact that that new type is supposed to be implemented somewhere and I just cloned Zotero's code, built it and it is not there.

Yes, the code is implemented in CSL, but it still needs to be added to Zotero, which is why the issue is still open with the `Zotero` label attached.

**Comment by bwiernik** [comment]
The best place to discuss future updates to Zotero fields is this thread https://forums.zotero.org/discussion/15636/changes-to-fields-and-item-types-for-zotero-5-1/p16

Zotero item type and field updates are underway. They will be announced when implemented

**Comment by dstillman** [comment]
https://forums.zotero.org/discussion/comment/471667/#Comment_471667

Since this is a new type, we can add this at any time with a full list of fields and CSL mappings.

**Comment by adam3smith** [comment]
The fields shouldn't be terribly difficult, but there's a general conceptual issue we need to agree on. People in the linked thread are actually asking about two different things: 1) Citing an entire journal and 2) Citing an entire journal _issue_ and possibly 3) something in between like an entire journal volume or a journal in a specific time span.

Do we think we can accommodate all of these with a single item type? I think we should be (and I think adding two item types is a bad idea), but we'll want to be clear that that's the goal.

**Comment by bwiernik** [comment]
We discussed this a bit when we added `periodical` in 1.0.2 and agreed that we could cover both with one item type. That's what's reflected in the spec https://docs.citationstyles.org/en/stable/specification.html#appendix-iii-types

When I see people wanting to cite a whole periodical (versus an issue), it's usually for something like a run of years or volumes (such as the tenure of an editor). 

I think both can be accommodated by including volume, issue, and container-title fields. Dates, volumes, and issues can all be ranges and so handle things like a period of years or editor tenure. For special issues, `title` would be the title of the issue and `container-title` would be the title of the journal. 

It could also potentially include `page` to handle special sections
